Sources/MapboxNavigation/PreviewViewControllerDelegate.swift

```diff
@@ -0,0 +1,26 @@
+import UIKit
+import CoreLocation
+import MapboxDirections
+
+// :nodoc:
+@_spi(Experimental) public protocol PreviewViewControllerDelegate: AnyObject {
+    
+    func previewViewControllerWillPreviewRoutes(_ previewViewController: PreviewViewController)
+    
+    func previewViewControllerWillBeginNavigation(_ previewViewController: PreviewViewController)
+    
+    func previewViewController(_ previewViewController: PreviewViewController,
+                               stateDidChangeTo state: PreviewViewController.State)
+    
+    func previewViewController(_ previewViewController: PreviewViewController,
+                               didLongPressFor coordinates: [CLLocationCoordinate2D])
+    
+    func previewViewController(_ previewViewController: PreviewViewController,
+                               didSelect route: Route)
+    
+    func previewViewController(_ previewViewController: PreviewViewController,
+                               bottomBannerViewControllerFor state: PreviewViewController.State) -> UIViewController?
```

Comment:
I don't like the idea of returning generic `UIViewController` here. Basically I'd like to have protocol constraints for specific `state`. E.g. in case of `.browsing` any `UIViewController` can be set, in case of `.destinationPreviewing` state any `UIViewController` that conforms to `DestinationPreviewing` typelias and so forth.

Any suggestions, @mapbox/navigation-ios?